### ISSUE TITLE
No way to disable __GETDS call generation on __interrupt

### ISSUE THREAD

**Issue by javispedro** [open-watcom/open-watcom-v2#845]
Since commit 6eb9ca014edccb478c3023a70481e8c3b83caccd a __GETDS call is always generated for __interrupt routines in 8086 . 
While in 386 where there is the "-zdl" option that disables __GETDS runtime calls (and instead forces a relocation), there is no such option in 8086. 

I have some code where the runtime is on another code segment (discarded) so the __GETDS call will fail. Examples:
A) A Win16 app where the C runtime is on a DISCARDABLE segment but the interrupt handler is by necessity on a FIXED one 
B) A DOS (.EXE) TSR where the C runtime was on the same segment but it is discarded after TSR, leaving only some interrupt handlers in memory.

I could force _GETDS proc in particular to be fixed/resident, but, as far as I can see, 
in both cases it is actually cheaper to just emit the relocation rather than the GETDS call.
I guess the GETDS call is only useful for targets without relocations at all e.g. .COM ?

This is actually a regression since in OW1.x these programs they work, as there are no __GETDS calls at all in 8086 mode.

I will try to make a patch to add the "-zdl" option to 8086 compiler too (once I find some time).
Filing this report in case there is some alternative setting I missed.

**Comment by jmalak** [comment]
It is not regression but wanted feature.
It is hard to decide the right solution because default differ between 16-bit and 32-bit compiler.
May be better solution is to use some new 16-bit option or use `zld` but with inverted meaning for 16-bit and 32-bit compiler.

**Comment by javispedro** [comment]
Default in both 32-bit compiler and 16-bit compiler (right now) is to generate __GETDS call.
-zdl flag just makes it do relocation instead of call.  No need to invert the meaning.

**Comment by jmalak** [comment]
You are not right, 16-bit compiler is now generating \_\_GETDS only for interrupt routines if small code model is used.
For other code direct relocation for DGROUP is used.

Default for 16-bit is relocation for DGROUP.
Default for 32-bit (which uses small/flat code model) is call \_\_GETDS routine and -zdl option switch behavior to use relocation for DGROUP (as 16-bit default).

**Comment by javispedro** [comment]
Yeah, you are right, the code is quite clear.

The way I saw the -zdl option was more like "_with -zdl never generate call to __GETDS routine, under any circumstances_". This is actually the meaning I'm interested in, because I want to avoid runtime calls. It's also easy to implement (just add the IsntTargetModel(LOAD_DS_DIRECTLY) condition as in the 32-bit compiler). 

However, i'm not sure if this is the most wanted meaning . Then again, as long as codegen can generate a __GETDS call (which it can after the above commit) , I think this option should exist. 

i.e. in 32-bit (current behavior):
| Code model | how to loadds without -zdl |with -zdl |
|  --------------- | ----------------- | ----------- |
| Any | call _GETDS | direct (relocation) |

while in 16 bit (proposed):
| Code model | loadds outside __interrupt with or without -zdl | loadds in __interrupt without -zdl | loads in __interrupt with -zdl |
| --- | --------------- | ----------------- | ----------- |
| Small | direct (relocation) | call _GETDS | direct (relocation) |
| Big | direct (relocation) | direct (relocation) | direct (relocation) |

**Comment by jmalak** [comment]
I return back old behavior and try to introduce tiny model based on cs segment register.
